### PR TITLE
[DRAFT] All Project Stein Changes So Far

### PR DESCRIPTION
**Pull request by anandagopal6**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs [hunk 1]
```diff
@@ -228,6 +228,7 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
 
             _workerChannelLogger.LogDebug("Received WorkerInitResponse. Worker process initialized");
             _initMessage = initEvent.Message.WorkerInitResponse;
+            _initMessage.Capabilities.Add("WorkerIndexing", "true"); // manually add capability for now without touching worker. Assume it is true.
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs]: This should be removed

File: src/WebJobs.Script.Grpc/Channel/WorkerCapabilities.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (soninaren) [inline src/WebJobs.Script.Grpc/Channel/WorkerCapabilities.cs]: Remove this file

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 3]
```diff
@@ -31,9 +32,10 @@ public class FunctionMetadataManager : IFunctionMetadataManager
         private ImmutableArray<FunctionMetadata> _functionMetadataArray;
         private Dictionary<string, ICollection<string>> _functionErrors = new Dictionary<string, ICollection<string>>();
         private ConcurrentDictionary<string, FunctionMetadata> _functionMetadataMap = new ConcurrentDictionary<string, FunctionMetadata>(StringComparer.OrdinalIgnoreCase);
+        private IFunctionMetadataProviderFactory _functionMetadataProviderFactory;
```
Comment (soninaren) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs]: nit: Factory could be readonly

File: src/WebJobs.Script/Host/FunctionMetadataProviderFactory.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            _applicationHostOptions = applicationHostOptions;
+            _metricsLogger = metricsLogger;
+            _loggerFactory = loggerFactory;
+        }
+
+        public void Create()
+        {
+            _workerFunctionMetadataProvider = new WorkerFunctionMetadataProvider(_applicationHostOptions, _loggerFactory.CreateLogger<WorkerFunctionMetadataProvider>(), _metricsLogger);
+
```
Comment (soninaren) [inline src/WebJobs.Script/Host/FunctionMetadataProviderFactory.cs]: nit: empty line

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 5]
```diff
@@ -76,11 +80,11 @@ public bool TryGetFunctionMetadata(string functionName, out FunctionMetadata fun
         /// <param name="applyAllowList">Apply functions allow list filter.</param>
         /// <param name="includeCustomProviders">Include any metadata provided by IFunctionProvider when loading the metadata</param>
         /// <returns> An Immmutable array of FunctionMetadata.</returns>
-        public ImmutableArray<FunctionMetadata> GetFunctionMetadata(bool forceRefresh, bool applyAllowList = true, bool includeCustomProviders = true)
+        public ImmutableArray<FunctionMetadata> GetFunctionMetadata(bool forceRefresh, bool applyAllowList = true, bool includeCustomProviders = true, IFunctionInvocationDispatcher dispatcher = null, bool workerIndexing = false)
```
Comment (soninaren) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs]: workerIndexing is unused

File: src/WebJobs.Script/Host/FunctionMetadataManager.cs [hunk 6]
```diff
@@ -117,14 +121,18 @@ private void InitializeServices()
         /// <summary>
         /// Read all functions and populate function metadata.
         /// </summary>
-        internal ImmutableArray<FunctionMetadata> LoadFunctionMetadata(bool forceRefresh = false, bool includeCustomProviders = true)
+        internal ImmutableArray<FunctionMetadata> LoadFunctionMetadata(bool forceRefresh = false, bool includeCustomProviders = true, IFunctionInvocationDispatcher dispatcher = null, bool workerIndexing = false)
```
Comment (soninaren) [inline src/WebJobs.Script/Host/FunctionMetadataManager.cs]: workerIndexing is unused

**Review by soninaren — CHANGES_REQUESTED** [review]
Requesting nit changes and others that we discussed on the call

**Comment by anandagopal6** [comment]
Closing this in favor of the larger PR with all Stein changes

**Pull request closed without merging**